sdk/spring/spring-cloud-azure-stream-binder-eventhubs/src/main/java/com/azure/spring/cloud/stream/binder/eventhubs/EventHubsMessageChannelBinder.java

```diff
@@ -225,7 +223,7 @@ private EventHubsTemplate getEventHubTemplate() {
             DefaultEventHubsNamespaceProducerFactory factory = new DefaultEventHubsNamespaceProducerFactory(
                 this.namespaceProperties, getProducerPropertiesSupplier());
 
-            clientFactoryCustomizers.forEach(customizer -> customizer.customize(factory));
+            producerFactoryCustomizers.forEach(customizer -> customizer.customize(factory));
```

Comment:
why do we design the interface to accept an argument of `abstract type` while in the implementation use its implementation? How about always using the implementation for clarity? By this way, we avoid the `is instanceof ` judgement then.

Reply:
We should use API instead of impl classes in APIs.